### PR TITLE
Get full sha for api repo in manifest file

### PR DESCRIPTION
**Pull request by utka**

### CODE REVIEW

File: release/gcb/generate_manifest.sh [hunk 1]
```diff
@@ -59,15 +59,19 @@ function create_manifest_check_consistency() {
     echo "ISTIO_REPO_SHA:$ISTIO_REPO_SHA API_REPO_SHA:$API_REPO_SHA PROXY_REPO_SHA:$PROXY_REPO_SHA CNI_REPO_SHA:$CNI_REPO_SHA some shas not found"
     exit 8
   fi
+  popd
+  checkout_code "api" "${API_REPO_SHA}" .
```
Comment (howardjohn) [inline release/gcb/generate_manifest.sh]: it's a bit confusing that the API_REPO_SHA is just the name of the directory and doesn't have any significance. Or did I understand it wrong?

Comment (utka) [reply release/gcb/generate_manifest.sh]: API_REPO_SHA is short sha which we get from go.mod file.
we use it to checkout api repo and get long sham which we can use in manifest file later on.

File: release/gcb/generate_manifest.sh [hunk 2]
```diff
@@ -59,15 +59,19 @@ function create_manifest_check_consistency() {
...
+  pushd api || exit
+    API_SHA=$(git rev-parse "${API_REPO_SHA}")
+  popd
```
Comment (fpesce) [inline release/gcb/generate_manifest.sh]: should we check this git rev-parse command error return somehow? (to detect if SHA is malformed or doesn't exist or something else?)

Comment (duderino) [reply release/gcb/generate_manifest.sh]: Wouldn't be bad, but either way we would see the errors in the logs.  So the error check would really just make it easier to debug a failure

**Pull request merged**